app/screens/global_threads/threads_list/thread/thread.tsx

```diff
@@ -0,0 +1,253 @@
+            backgroundColor: theme.buttonBg,
+            alignSelf: 'center',
+        },
+        mentionBadgeText: {
+            ...typography('Body', 25, 'SemiBold'),
+            alignSelf: 'center',
+            color: theme.buttonColor,
+        },
+    };
+});
+
+const Thread = ({author, channel, post, teammateNameDisplay, testID, thread}: Props) => {
+    const intl = useIntl();
+    const isTablet = useIsTablet();
+    const theme = useTheme();
+    const styles = getStyleSheet(theme);
+    const serverUrl = useServerUrl();
+
+    const handleMarkdownSoftBreak = useCallback(() => {
+        return '\n';
+    }, []);
+
+    const showThread = useCallback(preventDoubleTap(() => {
+        fetchAndSwitchToThread(serverUrl, thread.id);
+    }), [serverUrl, thread.id]);
+
+    const showThreadOptions = useCallback(() => {
+        const passProps = {thread};
+        const title = isTablet ? intl.formatMessage({id: 'thread.options.title', defaultMessage: 'THREAD ACTIONS'}) : '';
+
+        if (isTablet) {
+            showModal(Screens.THREAD_OPTIONS, title, passProps, bottomSheetModalOptions(theme, 'close-thread-options'));
+        } else {
+            showModalOverCurrentContext(Screens.THREAD_OPTIONS, passProps);
+        }
+    }, [isTablet, theme, thread]);
+
+    const threadStarterName = displayUsername(author, intl.locale, teammateNameDisplay);
+    const testIDPrefix = `${testID}.${thread.id}`;
+
+    const needBadge = thread.unreadMentions || thread.unreadReplies;
+    let badgeComponent;
+    if (needBadge) {
+        if (thread.unreadMentions) {
+            badgeComponent = (
+                <View
+                    style={styles.mentionBadge}
+                    testID={`${testIDPrefix}.unread_mentions`}
+                >
+                    <Text style={styles.mentionBadgeText}>{thread.unreadMentions > 99 ? '99+' : thread.unreadMentions}</Text>
+                </View>
+            );
+        } else if (thread.unreadReplies) {
+            badgeComponent = (
+                <View
+                    style={styles.unreadDot}
+                    testID={`${testIDPrefix}.unread_dot`}
+                />
+            );
+        }
+    }
+
+    let name;
+    let postBody;
+    if (post?.deleteAt && post.deleteAt > 0) {
+        name = (
+            <FormattedText
+                id='threads.deleted'
+                defaultMessage='Original Message Deleted'
+                style={[styles.threadStarter, styles.threadDeleted]}
+                numberOfLines={1}
+            />
+        );
+    } else {
+        name = (
+            <Text
+                style={styles.threadStarter}
+                numberOfLines={1}
+            >
+                {threadStarterName}
+            </Text>
+        );
+        if (post?.message) {
+            postBody = (
+                <Text
+                    style={styles.message}
+                    numberOfLines={2}
+                >
+                    <RemoveMarkdown
```

Comment:
@matthewbirtch is it to be expected that: 
1. emojis and mentions are stripped from markdown?
2. if the post has no text but it has other content like images or other attachments they do not show any content ?